fableous-fe/src/components/canvas/Canvas.tsx

```diff
@@ -261,15 +277,37 @@ const Canvas = forwardRef<HTMLCanvasElement, CanvasProps>(
       ctx.clearRect(0, 0, width, height);
       ctx.textAlign = "center";
       ctx.textBaseline = "middle";
-      Object.entries(textShapes).forEach(([id, shape]) => {
+      Object.entries(textShapesRef.current).forEach(([id, shape]) => {
         ctx.font = `${shape.fontSize * SCALE}px Arial`;
         ctx.fillText(
           shape.text,
           (shape.x1 + shape.x2) / 2,
           (shape.y1 + shape.y2) / 2
         );
-        if (parseInt(id, 10) === editingTextId) {
+        if (parseInt(id, 10) === editingTextIdRef.current) {
```

Comment:
what does the body of this conditional and the one with FRAME_COUNTER do?

Reply:
this highlights the texts in bright green boxes when a textShape is selected. the one with FRAME_COUNTER adds a blinking caret on controllers using ToolMode.Text -- just adding a signifier that they are to edit the selected text